### PR TITLE
chore(prismic): add NullContent documentation

### PR DESCRIPTION
**Pull request by paul-vd**
https://deploy-preview-428--heuristic-almeida-1a1f35.netlify.app/docs/prismic/resolver-cache.html#NullContent

### CODE REVIEW

File: source/docs/prismic/resolver-cache.md [hunk 1]
```diff
@@ -119,6 +119,61 @@ export default {
   **ProTip:** `contentProperty` can also be used with `singleton` Content.
 </blockquote>
 
+### NullContent
+
+The `NullContent` has been introduced to allow caching of resolvers that return `null`, lets say that you return a tag based on an id, but you only want to return the tags which are in the allow list, in this case, we will return `null` for the tags that are not in the allow list.
```
Comment (abailleulfc) [inline source/docs/prismic/resolver-cache.md:124]: split in two sentences : the feature and the example
```suggestion
The `NullContent` has been introduced to allow caching of resolvers that return `null`.

Lets say that you return a tag based on an id, but you only want to return the tags which are in the allow list, in this case, we will return `null` for the tags that are not in the allow list.
```

File: source/docs/prismic/resolver-cache.md [hunk 2]
```diff
@@ -119,6 +119,61 @@ export default {
...
+  Query: {
+    tag: PrismicCachedResolver((parent, args, { loaders }) => {
+      const tag = loaders.TagLoader.loadById(args.id);
+      return allowList.includes(tag.name) ? tag : null;
+    }),
+  },
+};
+```
+
+The issue we will have here is that there is no way creating a reference to the Tag document in prismic, so we won't be able to know that `tag3` or `tag4` should cache `null`, so we introduced the NullContent to handle this.
```
Comment (abailleulfc) [inline source/docs/prismic/resolver-cache.md:139]:
```suggestion
The issue we have here is that there is no way to create a reference to the Tag document in prismic, so we won't be able to know that `tag3` or `tag4` should cache `null`.

The NullContent handles this.
```

File: source/docs/prismic/resolver-cache.md [hunk 3]
```diff
@@ -119,6 +119,61 @@ export default {
...
+
+export default {
+  Query: {
+    tag: PrismicCachedResolver((parent, args, { loaders }) => {
+      const tag = loaders.TagLoader.loadById(args.id);
+      return allowList.includes(tag.name) ? tag : NullContent(tag);
+    }),
+  },
+};
+```
```
Comment (abailleulfc) [inline source/docs/prismic/resolver-cache.md:156]: I would say the most important here is the difference with the first code sample
```suggestion
```diff
import NullContent from "prismic/server/domain/NullContent";

const allowList = ["tag1", "tag2"];

export default {
  Query: {
    tag: PrismicCachedResolver((parent, args, { loaders }) => {
      const tag = loaders.TagLoader.loadById(args.id);
-      return allowList.includes(tag.name) ? tag : null;
+      return allowList.includes(tag.name) ? tag : NullContent(tag);
    }),
  },
};
```
```

File: source/docs/prismic/resolver-cache.md [hunk 4]
```diff
@@ -119,6 +119,61 @@ export default {
...
+      const tag = loaders.DocumentLoader.loadTagByDocumentID(
+        source.documentMetadata.documentID
+      );
+      return allowList.includes(tag.name)
+        ? tag
+        : NullContent(source.documentMetadata.documentID);
+    }),
+  },
+};
+```
```
Comment (abailleulfc) [inline source/docs/prismic/resolver-cache.md:175]: same here (either remove the allowList or add also the import) I would go for the smaller solution
```suggestion
...
export default {
  Fax: {
    tag: PrismicCachedResolver((parent, args, { loaders }) => {
      const tag = loaders.DocumentLoader.loadTagByDocumentID(
        source.documentMetadata.documentID
      );
      return allowList.includes(tag.name)
        ? tag
        : NullContent(source.documentMetadata.documentID);
    }),
  },
};
```
```

File: source/docs/prismic/resolver-cache.md [hunk 5]
```diff
@@ -119,6 +119,63 @@ export default {
...
+export default {
+  Query: {
+    tag: PrismicCachedResolver((parent, args, { loaders }) => {
+      const tag = loaders.TagLoader.loadById(args.id);
+-      return allowList.includes(tag.name) ? tag : null;
++      return allowList.includes(tag.name) ? tag : NullContent(tag);
+    }),
+  },
+};
+
```
Comment (abailleulfc) [inline source/docs/prismic/resolver-cache.md:161]: oh the diff didn't keep the closing backquotes

File: source/docs/prismic/resolver-cache.md [hunk 6]
```diff
@@ -119,6 +119,63 @@ export default {
...
+      const tag = loaders.DocumentLoader.loadTagByDocumentID(
+        source.documentMetadata.documentID
+      );
+      return allowList.includes(tag.name)
+        ? tag
+        : NullContent(source.documentMetadata.documentID);
+    }),
+  },
+};
+
```
Comment (abailleulfc) [inline source/docs/prismic/resolver-cache.md:178]: same here, the diff didn't keep the closing backquotes

**Review by abailleulfc — CHANGES_REQUESTED** [review]

**Comment by paul-vd** [comment]
closed in favor of https://github.com/front-commerce/developers.front-commerce.com/pull/431 which includes these changes

**Pull request closed without merging**